sdk/src/main/scala/com/horizen/SidechainState.scala

```diff
@@ -275,18 +313,34 @@ class SidechainState private[horizen] (stateStorage: SidechainStateStorage, val
     stateStorage.rollbackVersions.size
   }
 
-  override def rollbackTo(to: VersionTag): Try[SidechainState] = Try {
+  def rollbackTo(to: VersionTag, removedBlocks: Seq[SidechainBlock]): Try[SidechainState] = Try {
     require(to != null, "Version to rollback to must be NOT NULL.")
     val version = BytesUtils.fromHexString(to)
     applicationState.onRollback(version) match {
-      case Success(appState) => new SidechainState(stateStorage.rollback(new ByteArrayWrapper(version)).get, params, to, appState)
+      case Success(appState) => {
+        closedBoxesMerkleTree.removeBlocks(removedBlocks)
+        new SidechainState(stateStorage.rollback(new ByteArrayWrapper(version)).get, params, to, appState, closedBoxesMerkleTree)
```

Comment:
Maybe we need to do `stateStorage.rollback(new ByteArrayWrapper(version)).get` first, then ` closedBoxesMerkleTree.removeBlocks(removedBlocks)` and then new SidechainState creation.

Reply:
done.